pkg/network/state.go

```diff
@@ -989,7 +996,29 @@ func fixConnectionDirection(c *ConnectionStats) {
 	}
 	if sourceEphemeral && destNotEphemeral {
 		c.Direction = OUTGOING
-		stateTelemetry.udpDirectionFixes.Inc()
+		stateTelemetry.incomingDirectionFixes.Inc()
+	}
+}
+
+// fixOutgoingConnectionDirection potentially fixes the direction for outgoing connections
+//
+// When the system-probe starts up there is a race that occurs where port mappings are missing when they
+// are bound after the `/proc/net` state is read but before the probes are registered.  When that happens, the
+// connections will be marked as outgoing
+//
+// This function attempts to mitigate that by checking if an outgoing connection is from a non-ephemeral port to
+// an ephemeral port on an intra-host connection
+func fixOutgoingConnectionDirection(c *ConnectionStats) {
+	if c.Direction != OUTGOING || !c.IntraHost {
```

Comment:
nit: the direction check here is no longer needed.

Reply:
I'd prefer it as a guard against a future refactor as well as form of code documentation